src/js/view/controls/components/custom-button.ts

```diff
@@ -4,7 +4,7 @@ import svgParse from 'utils/svgParser';
 
 let collection = {};
 
-export function getCachedIcon(svg) {
+export function getCachedIcon(svg: string): HTMLElement {
```

Comment:
How about typing `collection`?

The [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) utility type is a great way to do this (especially if you want to type the svg name strings):

```ts
	let collection: Record<string, XMLDocument> = {};
```

or as the messier key-value pair:
```ts
	let collection: {[svg: string]: XMLDocument} = {};
```